cdap-ui/app/cdap/components/EntityListView/NoEntitiesMessage/__tests__/NoEntitiesMessage.test.js

```diff
@@ -0,0 +1,110 @@
+const filterEmptyMessagePath = 'features.EntityListView.emptyMessage.filter';
+const searchEmptyMessagePath = 'features.EntityListView.emptyMessage.search';
+let noEntitiesMessage;
+
+let filtersAreApplied = jest.fn();
+
+describe('NoEntitiesMessage Unit tests - ', () => {
+  it('Should render', () => {
+    filtersAreApplied.mockReturnValueOnce(false);
+    noEntitiesMessage = mount(
+      <NoEntitiesMessage
+        searchText=''
+        filtersAreApplied={filtersAreApplied}
+      />
+    );
+    expect(noEntitiesMessage.find('.empty-message-container').length).toBe(1);
+    expect(noEntitiesMessage.find('.empty-message-container strong').length).toBe(1);
+    expect(noEntitiesMessage.find('.empty-message-container .empty-message-suggestions').length).toBe(1);
+    noEntitiesMessage.unmount();
+  });
+
+  it('Should show appropriate default message', () => {
+    filtersAreApplied.mockReturnValueOnce(false);
+    noEntitiesMessage = mount(
+      <NoEntitiesMessage
+        searchText=''
+        filtersAreApplied={filtersAreApplied}
+      />
+    );
+    expect(noEntitiesMessage.find('.empty-message-container strong').text()).toBe(defaultEmptyMessagePath);
+    expect(noEntitiesMessage.find('.empty-message-container .action-item').length).toBe(1);
+    noEntitiesMessage.unmount();
+  });
+
+  it('Should show appropriate message when filters are applied', () => {
+    filtersAreApplied.mockReturnValueOnce(true);
+    noEntitiesMessage = mount(
+      <NoEntitiesMessage
+        searchText=''
+        filtersAreApplied={filtersAreApplied}
+      />
+    );
+    expect(noEntitiesMessage.find('.empty-message-container strong').text()).toBe(filterEmptyMessagePath);
+    expect(noEntitiesMessage.find('.empty-message-container .action-item').length).toBe(2);
+    noEntitiesMessage.unmount();
+  });
+
+  it('Should show appropriate message when there is a search query', () => {
+    filtersAreApplied.mockReturnValueOnce(false);
+    noEntitiesMessage = mount(
+      <NoEntitiesMessage
+        searchText='abc'
+        filtersAreApplied={filtersAreApplied}
+      />
+    );
+    expect(noEntitiesMessage.find('.empty-message-container strong').text()).toBe(searchEmptyMessagePath);
+    expect(noEntitiesMessage.find('.empty-message-container .action-item').length).toBe(2);
+    noEntitiesMessage.unmount();
+  });
+
+  it('Should show appropriate message when both filters and search query are applied', () => {
+    filtersAreApplied.mockReturnValueOnce(true);
+    noEntitiesMessage = mount(
+      <NoEntitiesMessage
+        searchText='abc'
+        filtersAreApplied={filtersAreApplied}
+      />
+    );
+    expect(noEntitiesMessage.find('.empty-message-container strong').text()).toBe(searchEmptyMessagePath);
+    expect(noEntitiesMessage.find('.empty-message-container .action-item').length).toBe(2);
+    noEntitiesMessage.unmount();
+  });
+
+  it('Should open up the Add Entity modal when "Add" is clicked', () => {
```

Comment:
If time permits we could add one more test that will create `NoEntitiesMessage` with `onSearch, onFiltersCleared ` function that gets called on clicking the Clear filter link.